mbuild/lib/recipes/polymer.py

```diff
@@ -100,28 +109,29 @@ def build(self, n, sequence='A'):
             if not tail[label].used:
                 tail_port = tail[label]
 
-        if not self.end_groups: # Cap each end with Hydrogens
+        if not self.end_groups and add_hydrogens: # Cap each end with Hydrogens
             hydrogen = H()
-            hydrogen['up'].update_separation(0.0547) # Defaut to H-C bond len
+            hydrogen['up'].update_separation(0.0547) # Defaut to 1/2 H-C bond len
             hydrogen_2 = clone(hydrogen)
             self.end_groups.extend([hydrogen, hydrogen_2])
-            head_port.update_separation(0.0547)
+            head_port.update_separation(0.0547) # 1/2 H-C bond len
             tail_port.update_separation(0.0547)
-        else: # Use compounds in self.end_groups
+        elif self.end_groups: # Use compounds in self.end_groups
```

Comment:
Does this need to be an `elif` or can it go in the `if` loop below?

Reply:
You're right, nice catch. Thanks!  I cleaned up this section a little bit in the latest commit.